### DISCUSSION TITLE
React to patches from immerjs's produce?

### DISCUSSION THREAD

**Discussion by geekox86** [solidjs/solid#1013] [Q&A]
Greetings,
I am wondering if it is possible to ask solidjs to directly react to patches (i.e. recorded changes) due to immerjs's produce without having to go through normal signals, stores, etc.?

In other words, can I write a library to utilize the core of solid and hook it directly to immerjs to facilitate reactivity?

Thx

**Comment by ryansolid** [comment]
Hmm.. I hadn't seen patches yet. It looks similar to overmind, so you could use them as a means to write to stores if you wanted. Maybe something like this that transforms ops: https://codesandbox.io/s/solid-overmind-5m8cs?file=/overmind.js

Signals are the core of Solid. Everything even rendering works off of them. So what you can do is take Immer's typical output and pass it into a store and run `reconcile` on it to basically diff it to get the changes back out of it. But honestly that's super inefficient since you have that information at the time of mutation and you are giving up and forcing a diff. While something like React welcomes this because they are a big diff engine it is opting into less performance innately.

Using patches is better, but alternatively you could use Solid's `produce` function with our stores and leverage the performance directly.

https://www.solidjs.com/docs/latest/api#produce

**Reply by geekox86** [reply]
Hi @ryansolid and thx for passing by. Do you mind if I try to understand Solid's source code better?
In what file does the actual interfacing between signals and rendering happen? I tried to look but I am missing the jsx source files imported in render/flow.js for example.
